vpr/src/server/serverupdate.cpp

```diff
@@ -0,0 +1,44 @@
+#ifndef NO_GRAPHICS
+
+#include "serverupdate.h"
+#include "gateio.h"
+#include "taskresolver.h"
+#include "globals.h"
+#include "ezgl/application.hpp"
+
+namespace server {
+
+gboolean update(gpointer data) {
+    bool isRunning = g_vpr_ctx.server().gateIO().isRunning();
+    if (isRunning) {
+        // shortcuts
+        ezgl::application* app = static_cast<ezgl::application*>(data);
+        GateIO& gate_io = g_vpr_ctx.mutable_server().mutable_gateIO();
+        TaskResolver& task_resolver = g_vpr_ctx.mutable_server().mutable_task_resolver();
+
+        std::vector<TaskPtr> tasksBuff;
+
+        gate_io.takeReceivedTasks(tasksBuff);
+        task_resolver.addTasks(tasksBuff);
+
+        bool process_task = task_resolver.update(app);
+
+        tasksBuff.clear();
+        task_resolver.takeFinished(tasksBuff);
+
+        gate_io.moveTasksToSendQueue(tasksBuff);
```

Comment:
I don't want to move entire buffer, because the gate_io might already have some tasks queue.
So i did moving task by task instead, hope this is fine.
resolved https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2526/commits/b2bd272052cf57d72a8cd1d663c57ceb795aa610